react/week4/context.js

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useState } from "react";
+
+export const StateContext = createContext();
+
+const StateProvider = (props) => {
+  const [loading, setLoading] = useState(false);
+  const [users, setUsers] = useState([]);
+
+  const fetchUsers = (query) => {
+    setLoading(true);
+    fetch(`https://api.github.com/search/users?q=${query}`)
```

Comment:
I suggest using async with try/catch here. It is more readable, imo

Reply:
I implemented it. Is it correct like this?